qiskit_nature/problems/second_quantization/molecular/hopping_ops_builder.py

```diff
@@ -0,0 +1,122 @@
+
+
+def _build_single_hopping_operator(index, num_particles, num_orbitals,
+                                   qubit_converter: QubitConverter):
+    h_1 = np.zeros((num_orbitals, num_orbitals), dtype=complex)
+    h_2 = np.zeros((num_orbitals, num_orbitals, num_orbitals, num_orbitals), dtype=complex)
+    z2_symmetries = qubit_converter.z2symmetries
+    if len(index) == 2:
+        i, j = index
+        h_1[i, j] = 4.0
+    elif len(index) == 4:
+        i, j, k, m = index
+        h_2[i, j, k, m] = 16.0
+    fer_op = fermionic_op_builder.build_ferm_op_from_ints(h_1, h_2)
+    qubit_op = qubit_converter.to_qubit_ops([fer_op])[0]  # TODO more elegant way than index
+    # if qubit_mapping == 'parity' and two_qubit_reduction:
+    #     qubit_op = TwoQubitReduction(num_particles=num_particles).convert(qubit_op[0])
+
+    commutativities = []
+    if not z2_symmetries.is_empty():
+        for symmetry in z2_symmetries.symmetries:
+            symmetry_op = PauliSumOp.from_list([(symmetry.to_label(), 1.0)])
+            commuting = qubit_op.primitive.table.commutes_with_all(
+                symmetry_op.primitive.table)
+            anticommuting = qubit_op.primitive.table.anticommutes_with_all(
+                symmetry_op.primitive.table)
+
+            if commuting != anticommuting:  # only one of them is True
+                if commuting:
+                    commutativities.append(True)
+                elif anticommuting:
+                    commutativities.append(False)
+            else:
+                raise QiskitNatureError(
+                    "Symmetry {} is nor commute neither anti-commute "
+                    "to exciting operator.".format(symmetry.to_label()))
+
+    return qubit_op, commutativities
+
+
+def build_hopping_operators(qmolecule: QMolecule, qubit_converter: QubitConverter,
+                            excitations: Union[str, List[List[int]]] = 'sd'
+                            ) -> Tuple[Dict[str, PauliSumOp],
+                                       Dict[str, List[bool]],
+                                       Dict[str, List[Any]]]:
+    """Builds the product of raising and lowering operators (basic excitation operators)
+
+    Args:
+        excitations: The excitations to be included in the eom pseudo-eigenvalue problem.
+            If a string ('s', 'd' or 'sd') then all excitations of the given type will be used.
+            Otherwise a list of custom excitations can directly be provided.
+
+    Returns:
+        A tuple containing the hopping operators, the types of commutativities and the
+        excitation indices.
+    """
+
+    num_alpha, num_beta = qmolecule.num_alpha, qmolecule.num_beta
+    num_molecular_orbitals = qmolecule.num_molecular_orbitals
+    num_orbitals = 2 * num_molecular_orbitals
+
+    if isinstance(excitations, str):
+        se_list, de_list = UCCSD.compute_excitation_lists([num_alpha, num_beta], num_orbitals,
```

Comment:
This `UCCSD` class will be removed. In fact, the whole `qiskit_nature.components` module will be removed.
You should be able to leverage the new code in `qiskit_nature.circuit.library.ansatzes`. Especially the excitation generator code from `qiskit_nature.circuit.library.ansatzes.utils.fermionic_excitation_generator`